Added an example on how to integrate Matplotlib

SGCODEX opened the pull request:
This example shows how to create a Matplotlib chart and how to integrate it in a Taipy page using the figure property.
Screenshot of Taipy page: 
![Example SS](https://github.com/user-attachments/assets/a24ae074-362d-4493-8332-f81b4d5b388f)

fixes #558
Please review @jrobinAV sir
Thank you

SGCODEX committed:
Added example on how to integrate Matplotlib

Using the figure property

FabienLelaquais reviewed (dismissed):
Excellent!
Simple and effective. Well done!

Now we will have to move this at another location, and add a Markdown version of this.
And add a little piece of doc referring to this and explain what's going on.
But that's an easy shot that I'll take care of.

Thank you!

SGCODEX commented:
Thank you sir @FabienLelaquais
Sure sir, and please let me know if I can also help more in this issue.
Also sir, will this PR be merged after the changes?

Thank you sir

FabienLelaquais commented:
This PR will certainly be merged, yes, @SGCODEX!
Hacktoberfest PRs are not merged right away though, because of the rewarding process that goes on.
But I promise I'll let you know when this gets is (a matter of days) so we all celebrate!

Thanks again

SGCODEX commented:
Thank you so much sir,
Looking forward towards the celebration and more meaningful contributions. 😇

FredLL-Avaiga reviewed (changes requested):
Can you fix the linter issues please ?
And update your branch ?
Tx @SGCODEX

SGCODEX committed:
Merge branch 'develop' into develop

SGCODEX commented:
@FredLL-Avaiga Sir,
I have updated the branch.
Can you tell me about linter issues? I don't know about them. Please help sir on what to fix.

FlorianJacta commented on doc/gui/examples/matplotlib_example.py:
```diff
@@ -0,0 +1,22 @@
+# Integrating Matplotlib into Taipy using figure property
```
Where are you using the figure property?

SGCODEX replied:
Sir, I am using matplotlib.figure property as this class is automatically recognized by Taipy GUI and can be passed to the content property. Thus the Matplotlib output can be inserted in the part block using figure property.

FlorianJacta replied:
This is a bit confusing as Taipy has a figure property for its charts. What about: # Integrating Matplotlib Figure into Taipy

FabienLelaquais replied:
It is confusing indeed and I plan to fix that when I'm done with (far) more urgent tasks.
This really is a demo of how to use the part block to hold a Matplotlib figure. That's where it will go, and the chart control documentation will reference that.

FlorianJacta replied:
I am fine with that. I am confused by the wording above and propose one that is less confusing in my opinion

SGCODEX replied:
I understand sir. Have removed "# Integrating Matplotlib Figure into Taipy" for now. If there can be a comment which depicts it better, we can go ahead with that sir. Thank you sir.

FlorianJacta commented on doc/gui/examples/matplotlib_example.py:
```diff
@@ -0,0 +1,22 @@
+from taipy.gui import Gui 
+import taipy.gui.builder as tgb
+import matplotlib.pyplot as plt  
+from matplotlib.figure import Figure 
```
Is `Figure` used in the code? Why do you import it?

SGCODEX replied:
Right sir, the matplotlib.figure.Figure property is not being used here. Will make the changes in updated code. Thank you sir.

FlorianJacta reviewed (commented):
Little comments, but otherwise, great!

FredLL-Avaiga commented:
> @FredLL-Avaiga Sir, I have updated the branch. Can you tell me about linter issues? I don't know about them. Please help sir on what to fix.

![image](https://github.com/user-attachments/assets/baa58d79-3427-46ac-a2c6-81a4dcc528f1)

Click on the [details link](https://github.com/Avaiga/taipy/actions/runs/11216526299/job/31179383498?pr=1930)

SGCODEX commented:
@FredLL-Avaiga @FlorianJacta Sir, working on these issues. Will submit the commits solving them soon. Thank you sirs.

SGCODEX committed:
Merge branch 'Avaiga:develop' into develop

SGCODEX committed:
fixed linter issues

FredLL-Avaiga commented:
here is your code with the import sorted
```
import matplotlib.pyplot as plt
import numpy as np

import taipy.gui.builder as tgb
from taipy.gui import Gui

fig = plt.figure(figsize=(5, 4))
xx = np.arange(0, 2 * np.pi, 0.01)
plot = fig.subplots(1, 1)
plot.fill(xx, np.sin(xx), facecolor="none", edgecolor="purple", linewidth=2)

# Create a Taipy Page
with tgb.Page() as page:
    tgb.part(content="{fig}", height="520px")

# Run the Taipy Application:
if __name__ == "__main__":
    Gui(page=page).run(title="Matplotlib Example")

```

SGCODEX committed:
Updated Code

SGCODEX committed:
Merge branch 'Avaiga:develop' into develop

SGCODEX committed:
Merge branch 'develop' of https://github.com/SGCODEX/taipy into develop

SGCODEX commented:
Thank you @FredLL-Avaiga Sir, Have updated the code.

SGCODEX commented:
@FredLL-Avaiga please review sir

FredLL-Avaiga reviewed (changes requested):
your branch is out of date, please update it
@FabienLelaquais had requested another example identical but using the Markdown syntax, Can you add it ?

SGCODEX commented:
@FredLL-Avaiga Sir, Sure I ll update the branch and write the same code in Markdown File.

SGCODEX committed:
Merge branch 'Avaiga:develop' into develop

SGCODEX committed:
Added Matplotlib_Example Markdown File

SGCODEX committed:
Merge branch 'Avaiga:develop' into develop

SGCODEX commented:
@FredLL-Avaiga Sir,
Added the requested Markdown syntax file sir, with another example. Also updated the branch.
Output of this example:
![MEOutput](https://github.com/user-attachments/assets/648847b0-b8ff-4df1-b2f7-5ade4ff50257)

FredLL-Avaiga commented on doc/gui/examples/matplotlib_example1.md:
I think we might not have been clear enough. 
When we mentioned a markdown example, we meant a python example that uses the markdown syntax for describing the page instead of the taipy.gui.builder syntax.

SGCODEX replied:
Hello sir, I understand.. working on this again.

FredLL-Avaiga committed:
Merge branch 'develop' into develop

FredLL-Avaiga commented:
Are you still working on this PR @SGCODEX  ?
If you are, please answer the questions
If you are not, please close the PR

SGCODEX commented:
I am working on it sir @FredLL-Avaiga

SGCODEX committed:
Merge branch 'Avaiga:develop' into develop

SGCODEX committed:
Updated matplotlib examples

SGCODEX commented:
Hello @FredLL-Avaiga Sir and @FabienLelaquais Sir
I have added two new examples using markdown syntax or taipy.gui/Gui syntax. It was hard to incorporate the entire code of matplotlib into the markdown syntax as this syntax either takes the values as data and then plots it using chart, thus eliminating the need of matplotlib completely. But as I was asked to give example with matplotlib, after trying a lot of different ways I made two examples, one is simple code which takes values as data and plots as charts. **Other one is more specific to what is required, where I save the matplotlib graph generated as an image and then show that image instead.**

Screenshots:
Example 1)
![image](https://github.com/user-attachments/assets/0c5bbd71-e61c-4130-85ff-7926287ff9f9)

Example 2)
![Screenshot 2024-10-29 014623](https://github.com/user-attachments/assets/b91a9f60-08cf-4b50-bc5f-989db13aa305)

Sir, I understand while this may not be exactly how you wanted but please guide me and help me to make the changes as required and make this a perfect example. Also, would really love to solve this issue finally and get this PR merged and successfully contribute to hacktoberfest under taipy. Please help me solve it and make this a meaningful contribution.

Thank you

SGCODEX commented:
@FredLL-Avaiga Sir, please let me know if the Markdown part of this PR is not required. I will just change it back to the tgb part and then this PR can be merged.

SGCODEX commented:
> @FredLL-Avaiga Sir, please let me know if the Markdown part of this PR is not required. I will just change it back to the tgb part and then this PR can be merged.

Also, please let me know sir what else needs to be done to get this merged as I have been contributing since the start of Hacktoberfest under this issue and would really liked it to be merged asap, before this month ends. Cc: @FabienLelaquais sir

FredLL-Avaiga commented:
> > @FredLL-Avaiga Sir, please let me know if the Markdown part of this PR is not required. I will just change it back to the tgb part and then this PR can be merged.
> 
> Also, please let me know sir what else needs to be done to get this merged as I have been contributing since the start of Hacktoberfest under this issue and would really liked it to be merged asap, before this month ends. Cc: @FabienLelaquais sir

no worries about the date, you're in

FredLL-Avaiga committed:
Merge branch 'develop' into develop

FredLL-Avaiga commented on doc/gui/examples/matplotlib/markdown_example1.py:
this is not a matplotlib example, can you remove it ?

SGCODEX replied:
Removed Sir

FredLL-Avaiga commented on doc/gui/examples/matplotlib/images/graph.png:
remove this

SGCODEX replied:
Removed Sir

FredLL-Avaiga commented on doc/gui/examples/matplotlib/example.md:
remove this

SGCODEX replied:
Removed Sir

FredLL-Avaiga commented on doc/gui/examples/matplotlib/markdown_matplotlib_example2.py:
```diff
@@ -0,0 +1,23 @@
+# Define a page using Markdown syntax
+page_content = """
+# Taipy Example for Matplotlib Integration
+<|{file_path}|image|>
```
we don't want to use image but <|part|content={fig}|>

SGCODEX replied:
Have used <|part|content={fig}|class_name=matplotlib_example|> now sir

FredLL-Avaiga commented on doc/gui/examples/matplotlib/tgb_matplotlib_example.py:
```diff
@@ -0,0 +1,19 @@
+
+# Create a Taipy Page
+with tgb.Page() as page:
+    tgb.part(content="{fig}", height="520px")
```
👍

FredLL-Avaiga reviewed (changes requested):
please remove the non related files and fiw the markdown_matplotib_example.py
and fix the linter issues

SGCODEX committed:
Updated markdown example

SGCODEX commented:
Greetings @FredLL-Avaiga Sir,
Have made the required changes, please review.
SS:
![image](https://github.com/user-attachments/assets/530277a9-7507-4314-899a-f76df6a897f4)
Thank you

SGCODEX committed:
fixing linter issues

FredLL-Avaiga commented on doc/gui/examples/matplotlib/markdown_matplotlib_example.py:
```diff
@@ -0,0 +1,25 @@
+# Define a page using Markdown syntax
+page_content = Markdown("""
+# Taipy Example for Matplotlib Integration using Markdown Syntax
+<|part|content={fig}|class_name=matplotlib_example|>
```
👍

FredLL-Avaiga commented on doc/gui/examples/matplotlib/markdown_matplotlib_example.py:
```diff
@@ -0,0 +1,25 @@
+#Using markdown syntax
```
We need the copyright

FredLL-Avaiga commented on doc/gui/examples/matplotlib/tgb_matplotlib_example.py:
```diff
@@ -0,0 +1,19 @@
+#Using taipy.gui.builder syntax
```
Copyright

FredLL-Avaiga reviewed (changes requested):
Is it not possible to have the exact same example in markdown and builder? 
If it is, change the structure to a directory with 
- markdown.py
- builder.py

You can check out the other directory at the same level

SGCODEX committed:
Merge branch 'Avaiga:develop' into develop

SGCODEX committed:
Made examples similar and added Copyright

SGCODEX commented:
@FredLL-Avaiga Sir,
Have made both the examples exactly similar to each other. Added copyright to both the files.
Screenshot of both the file's output:
![image](https://github.com/user-attachments/assets/6bffd599-fe72-4eaa-ab73-aba93c6b2cf6)

SGCODEX commented:
> Is it not possible to have the exact same example in markdown and builder? If it is, change the structure to a directory with
> 
> * markdown.py
> * builder.py
> 
> You can check out the other directory at the same level

I have tried to rename the files to markdown.py and builder.py sir in a directory "matplotlib", but I think taipy has some other file named as builder.py as I was getting this circular import error:
![image](https://github.com/user-attachments/assets/3ee69326-d2c8-43b4-a499-3833e945f7d0)

So have renamed like this:
matplotlib
* builder_example.py
* markdown_example.py

FredLL-Avaiga commented on doc/gui/examples/matplotlib/builder_example.py:
```diff
@@ -0,0 +1,37 @@
+plot = fig.subplots(1, 1)
+plot.fill(xx, np.sin(xx), facecolor="none", edgecolor="purple", linewidth=2)
+
+with tgb.Page() as page:
```
define the style in `tgb.Page()`

FredLL-Avaiga reviewed (changes requested):
Can you organise the files like this example ?
https://github.com/Avaiga/taipy/blob/5f982e1a3671a430fe42e1759401eeb0e510db2d/doc/gui/examples/controls/progress_styling_circular/

SGCODEX committed:
Updated file structure & added style to builder.py

SGCODEX commented:
@FredLL-Avaiga Sir, Have organized the files like the given example and defined the style in tgb.Page()

FredLL-Avaiga commented on doc/gui/examples/charts/matplotlib/builder.py:
```diff
@@ -0,0 +1,44 @@
+            }
+        }
+) as page:
+    tgb.text(value="# Taipy Example for Matplotlib Integration", mode="md")
```
I guess this should be `tgb.html("h1", "Taipy Example for Matplotlib Integration")`

FredLL-Avaiga commented on doc/gui/examples/charts/matplotlib/builder.py:
```diff
@@ -0,0 +1,44 @@
+
+# Run the Taipy Application:
+if __name__ == "__main__":
+    Gui(page=page).run(title="Matplotlib Example")
```
`Gui(page).run(...`

FredLL-Avaiga reviewed (changes requested):
a last comment

SGCODEX committed:
__init__.py

SGCODEX committed:
Updated files

SGCODEX commented:
Done sir @FredLL-Avaiga

SGCODEX committed:
Trying to fix linter import issues

SGCODEX committed:
Trying to fix linter import issues

SGCODEX commented:
@FredLL-Avaiga Not understanding how to fix the import linter issues. Please help sir.

FredLL-Avaiga commented on doc/gui/examples/charts/matplotlib/builder.py:
```diff
@@ -0,0 +1,44 @@
+# installed.
+# -----------------------------------------------------------------------------------------
+# Matplotlib example
+import numpy as np
+import matplotlib.pyplot as plt
+
+from taipy.gui import Gui
+import taipy.gui.builder as tgb
```
```
import numpy as np

import matplotlib.pyplot as plt
import taipy.gui.builder as tgb
from taipy.gui import Gui
```

FredLL-Avaiga commented on doc/gui/examples/charts/matplotlib/markdown.py:
```diff
@@ -0,0 +1,43 @@
+# installed.
+# -----------------------------------------------------------------------------------------
+# Matplotlib example
+import numpy as np
+import matplotlib.pyplot as plt
+
+from taipy.gui import Gui, Markdown
```
```
import numpy as np

import matplotlib.pyplot as plt
from taipy.gui import Gui, Markdown
```

SGCODEX committed:
fixed import issues

SGCODEX commented:
Thank you sir @FredLL-Avaiga

SGCODEX commented:
Anything else remains in this sir? Or can this be merged?

SGCODEX commented:
@FredLL-Avaiga Please review sir. Thank you.

FredLL-Avaiga committed:
Merge branch 'develop' into develop

FredLL-Avaiga commented:
can you resolves the GitHUb discussions, once you answered them ?

FredLL-Avaiga reviewed (approved):
excellent job @SGCODEX

FredLL-Avaiga commented:
@jrobinAV Can you merge this PR ?

jrobinAV commented:
Rock on !!

SGCODEX commented:
Thank you sirs @FredLL-Avaiga @jrobinAV
(Sure, I will resolve the Github discussions too)

RymMichaut commented:
Congratulations @SGCODEX 🥳🦾
Please contact me at rym.michaut@taipy.io to prepare and send your swag 🎁

SGCODEX commented:
Thank you ma'am @RymMichaut 😊😊
Mailed ma'am!!
